sw/host/rom_ext_image_tools/meson.build

```diff
@@ -0,0 +1,65 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+cargo = find_program('cargo', required: false, disabler: true)
+
+build_type = 'release'
+build_dir = meson.current_build_dir()
+manifest_path = meson.current_source_dir() / 'Cargo.toml'
+toolchain_file = meson.current_source_dir() / 'rust-toolchain'
+
+# CARGO FLAGS:
+# These flags will only apply to the final binary, and won't get propogated
+# to the dependency crate builds.
+cargo_flags_array = [
+  'manifest-path=' + manifest_path,
+  'target-dir=' + build_dir,
+   build_type,
+]
+
+cargo_flags = ''
+foreach flag : cargo_flags_array
+  cargo_flags += '--' + flag + ' '
+endforeach
+
+# RUSTFLAGS:
+# These flags will apply to all the dependencies, as well as the final
+# binary. Linker and linker flavor amongst other things can be passed through
+# these flags.
+rust_flags = ''
```

Comment:
For Follow-up: I'm worried by these two - and how this is dealt with in `invoke_cargo.sh`. I think we can find a better way, but that should not block this PR